File: frontend/src/api/archibus.ts
import { api } from 'src/api/http';

export interface ArchibusBuilding {
  building_location: string;
  building_name: string;
}

export interface ArchibusRoom {
  unit_institutional_id: string | null;
  building_location: string;
  building_name: string;
  room_name: string;
  room_type: string | null;
  room_surface_square_meter: number | null;
  heating_kwh_per_square_meter: number | null;
  cooling_kwh_per_square_meter: number | null;
  ventilation_kwh_per_square_meter: number | null;
  lighting_kwh_per_square_meter: number | null;
}

export async function getArchibusBuildings(
  unitId: number,
): Promise<ArchibusBuilding[]> {
  return api
    .get('modules/archibus-rooms', {
      searchParams: { unit_id: unitId },
    })
    .json<ArchibusBuilding[]>();
}

export async function getArchibusRooms(
  unitId: number,
  buildingName: string,
): Promise<ArchibusRoom[]> {
  return api
    .get('modules/archibus-rooms', {
      searchParams: { unit_id: unitId, building_name: buildingName },
    })
    .json<ArchibusRoom[]>();
}

File: backend/app/schemas/archibus.py
"""Archibus API response schemas."""

from typing import Optional

from pydantic import BaseModel, ConfigDict


class ArchibusBuildingResponse(BaseModel):
    building_location: str
    building_name: str


class ArchibusRoomResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    unit_institutional_id: Optional[str] = None
    building_location: str
    building_name: str
    room_name: str
    room_type: Optional[str]
    room_surface_square_meter: Optional[float]
    heating_kwh_per_square_meter: Optional[float]
    cooling_kwh_per_square_meter: Optional[float]
    ventilation_kwh_per_square_meter: Optional[float]
    lighting_kwh_per_square_meter: Optional[float]
